gunfolds/utils/clingo.py

```diff
@@ -41,26 +41,45 @@ def run_clingo(command,
         trendy: Use defaults geared towards industrial problems
     :type configuration: string
 
-    :param cpath: clingo path
-    :type cpath: string
 
     :param pnum: number of parallel threads to run clingo on
     :type pnum: integer
 
+    :param optim: a list containing configuration for optimization algorithm and optionally a bound  [<arg>, <bound>]
+        <arg>: <mode {opt|enum|optN|ignore}>[,<bound>...]
+        opt   : Find optimal model
+        enum  : Find models with costs <= <bound>
+        optN  : Find optimum, then enumerate optimal models
+        ignore: Ignore optimize statements
+        <bound> : Set initial bound for objective function(s)
+    :type optim: list
+
     :returns: results of equivalent class
     :rtype: dictionary
     """
+    if optim is None:
+        optim = ['optN']
     if pnum is None:
         pnum = PNUM
-    ctrl = clngo.Control(["--warn=no-atom-undefined","--configuration=", configuration, "-t", str(int(pnum)) + ",split", "-n", str(capsize)])
+    if len(optim) == 1:
+        optim_option = optim[0]
+    elif len(optim) == 2:
+        optim_option = f"{optim[0]}, {optim[1]}"
```

Comment:
let's replace lines 64-67 with this single line?
`optim_option = ", ".join([x for x in optim])`

Reply:
well, maybe it will not e able to catch cases of users passing more than 2 :(

Reply:
I solved this above in a much simpler way